outlines/generate/processors.py

```diff
@@ -0,0 +1,78 @@
+import math
```

Comment:
Great job creating generic logits processors! This will help a lot with applying logits processors to other inference engines!

Reply:
I realized llama.cpp and vLLM do not both pass the prompt token ids in the logit biasing function. I am thus moving these classes from `outlines.generate`  to `outlines.models.llamacpp`